src/main/java/com/aws/greengrass/clientdevices/auth/certificate/infra/ClientCertificateStore.java

```diff
@@ -76,12 +82,28 @@ public void storePem(String certificateId, String certificatePem) throws Certifi
         keyStore.setCertificateEntry(certificateId, cert);
     }
 
+    /**
+     * Removes the PEM for a certificateId alias.
+     * @param certificateId - a certificate id
+     */
+    public void removePem(String certificateId) {
+        try {
+            keyStore.deleteEntry(certificateId);
+        } catch (KeyStoreException e) {
+            logger.atError().cause(e).kv("certificateId", certificateId).log("Failed to remove certificate PEM");
```

Comment:
when would this happen? If it does happen, is there anything the customer can do about it?

Reply:
Based on the docs it could happen if the entry can't be removed or if the keystore has not been initialized. This is not supper critical as this keystore lives in memory and it might happen you that we try to delete a certificate but the alias is not in there.

I guess we could make the case for throwing the exception and letting the client handle it

Reply:
OK, maybe go for WARN here if there's nothing the customer can do about it.